packages/firestore/src/core/sync_engine.ts

```diff
@@ -949,11 +950,12 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
         );
         // For queries that never executed on this client, we need to
         // allocate the query in LocalStore and initialize a new View.
-        const query = await this.localStore.getQueryForTarget(targetId);
-        assert(!!query, `Query data for target ${targetId} not found`);
-        queryData = await this.localStore.allocateQuery(query!);
+        const target = await this.localStore.getTarget(targetId);
+        assert(!!target, `Target for id ${targetId} not found`);
+        queryData = await this.localStore.allocateTarget(target!);
         await this.initializeViewAndComputeSnapshot(
-          queryData,
+          target!.toQuery(),
```

Comment:
I am not sure converting a query to a target here will always work. If a secondary tab issues a `limitToLast` query, and the primary tab executes the query without knowing this, then the computed View is likely different. Unfortunately, we use the View to determine which documents to return for Limit queries (https://github.com/firebase/firebase-js-sdk/blob/5beb23cd47312ffc415d3ce2ae309cc3a3fde39f/packages/firestore/src/core/view.ts#L226). I think executing the "wrong query" might break Limbo resolution.